### PR TITLE
add ContextAction for redo

### PR DESCRIPTION
**Pull request by kendonB**
Addresses https://github.com/dictation-toolbox/Caster/issues/426.

note that this requires updating dragonfly2 to the latest version.

- [x] @kendonB test exact branch on home computer, including repeats
- [x] Someone else test it too with rstudio and one other app, including repeats

### CODE REVIEW

**Comment by kendonB** [comment]
Note that next time I will create the new branch on my own fork. I just need to work out how to do that

**Comment by kendonB** [comment]
Looks like the repeats aren't working. @LexiconCode, can you see what I've done wrong?

**Comment by LexiconCode** [comment]
I'll see if I get this tested tomorrow!

**Comment by LexiconCode** [comment]
This works as expected. You can add the bolded applications in #426.

File: castervoice/lib/ccr/core/nav.py [hunk 1]
```diff
@@ -95,7 +96,10 @@ class NavigationNon(MappingRule):
         "undo [<n>]":
             R(Key("c-z"), rdescript="Core: Undo")*Repeat(extra="n"),
         "redo [<n>]":
-            R(Key("c-y"), rdescript="Core: Redo")*Repeat(extra="n"),
+            R(ContextAction(default=Key("c-y")*Repeat(extra="n"), actions=[
+                # Use cs-z for rstudio
+                (AppContext(executable="rstudio"), Key("cs-z")*Repeat(extra="n")),
```
Comment (lexxish) [inline castervoice/lib/ccr/core/nav.py]: This should be in app grammar. Jetbrains is also cs-z for example

Comment (kendonB) [reply castervoice/lib/ccr/core/nav.py]: Using contextaction is an alternative to using app grammars.

Comment (kendonB) [reply castervoice/lib/ccr/core/nav.py]: Will add jetbrains. Does c-y not work there?

Comment (lexxish) [reply castervoice/lib/ccr/core/nav.py]: I meant from a design perspective since otherwise the code will get polluted with app specific if statements

Comment (kendonB) [reply castervoice/lib/ccr/core/nav.py]: In this case where there are just a few exceptions to the common key mapping, I think I prefer ContextAction to adding the exceptions to App grammars. Let's see how it goes doing it this way and if the core grammars start to engorge we can move stuff into app files.

Comment (kendonB) [reply castervoice/lib/ccr/core/nav.py]: @lexxish, The apps can be specified as a list as well, so adding a new app to a ContextAction is super compact. Now that I know this, I'm skeptical that it's better to have app grammars at all.

**Pull request merged**